### PR TITLE
Added showNA property to generic assay charts

### PR DESCRIPTION
**Pull request by dianab0**
ShowNAs is available for generic assay charts but not working.

Replication steps:
- Go to ccle_broad_2019 study
- Select Chart -> Treatment Response -> Pick a treatment and Add Chart
- Then go to the displayed histogram, hover the top-right corner, and pick "Show NAs"
- It is not working and there will be the following error in the console:
ShowNAError

ShowNA_Error

The fix can be checked here: https://cbioportal.thehyve.net/study/summary?id=ccle_broad_2019 using account testuser - P@ssword.

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -352,22 +352,49 @@ export class StudyViewPageStore {
...
     };
 
     public isShowNAChecked = (uniqueKey: string): boolean => {
-        let filter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey));
+        let filter;
+        if (this.isGenericAssayChart(uniqueKey))
+            filter = _.clone(
+                this._genericAssayDataBinFilterSet.get(uniqueKey)
+            )!;
+        else filter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey));
```
Comment (dippindots) [inline src/pages/studyView/StudyViewPageStore.ts:396]: Personally I prefer a full brace style, but I am ok with this also.

**Review by dippindots — APPROVED** [review]
Hi @dianab0, thanks for fixing this issue! It looks good to me.

File: src/pages/studyView/StudyViewPageStore.ts (hunk 1 above)
Comment (alisman) [reply src/pages/studyView/StudyViewPageStore.ts:396]: @dianab0 lets just put in the full braces.  want to keep style consistency

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -352,22 +352,49 @@ export class StudyViewPageStore {
...
     };
 
     @action
     toggleNAValue = (uniqueKey: string): void => {
         this.updateNAValue(uniqueKey);
-        let newFilter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey))!;
-        let showNA = newFilter.showNA;
-        newFilter.showNA = !showNA;
-        this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        if (this.isGenericAssayChart(uniqueKey)) {
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:373]: seems this code is quite redundant.   how about:
```
targetWhateverSet = this.isGenericAssChart(uniqueKey);
// now proceed with cloning and setting
```

Comment (dianab0) [reply src/pages/studyView/StudyViewPageStore.ts:380]: I tried this but I get an error when trying to set newFilter to targetWhateverSet (targetWhateverSet.set(uniqueKey, newFilter)) as newFilter can have two types GenericAssayDataBinFilter or ClinicalDataBinFilter.
However, I modified a bit the method to be less redundant.

**Comment by dianab0** [comment]
Thank you for review! I updated the code under the same commit.

**Comment by alisman** [comment]
@adamabeshouse does this duplicate the work you did?

**Comment by alisman** [comment]
@adamabeshouse can you decide we decide which to merge (if i'm right that it's redundant)

**Comment by adamabeshouse** [comment]
@alisman ah, I didn't realize this was a problem for generic assay charts too. this does not duplicate but it would dovetail well with my work. lets merge this and then Ill combine it in my PR

**Pull request merged**